src/reader.rs

```diff
@@ -1627,6 +1696,50 @@ impl<Term: Terminal> Reader<Term> {
         Ok(())
     }
 
+    /// Checks whether an receiver exists.
+    fn has_receiver(&mut self) -> bool {
+        self.log_channel.is_some()
+    }
+
+    /// Gets the next line in the receiver if one exists. Otherwise, it returns `None`.
+    fn receive_next(&mut self) -> Option<String> {
+        let raw_recv = self.log_channel.as_ref().and_then(|c| c.receive());
+        if raw_recv.is_none() && self.log_channel.as_mut().map_or(false, |x| x.is_dead()) {
+            self.log_channel = None
```

Comment:
A little nitpicking: This line should end with a semicolon. It's not strictly necessary, but I think it makes it clearer that the expression isn't being yielded up one block.

The same for lines 1724, 1733, 2254, 2256, 2259, 2268, and 2771.